fhireval/test_suite/crud/__init__.py

```diff
@@ -1,83 +1,36 @@
 test_set_id = "2.5"
 
 # User friendly name of the tests
 test_set_name = "CRUD"
 
+import pytest
+from fhir_walk.model import unwrap_bundle
+
+# This is the function that we'll use to grab the JSON files
+from fhireval.data import load_test_data, delete_content_for_resource
+
+import pdb
+
 # Probably best for tests to log to this one so that they will all have the same entry
 import logging
+
 logger = logging.getLogger(__name__)
 
-# This will be set by the create test
-example_patient_id = None
-example_patient = {
-  "resourceType": "Patient",
-  "meta": {
-    "profile": [
-      "http://hl7.org/fhir/StructureDefinition/Patient"
-    ]
-  },
-  "identifier": [
-    {
-      "system": "https://ncpi-api-dataservice.kidsfirstdrc.org/participants?study_id=FAKE-CMG&external_id=",
-      "value": "ctpatient01"
-    },
-    {
-      "system": "urn:ncpi:unique-string",
-      "value": "CRUDTEST|ctpatient01"
-    }
-  ],
-  "extension": [
-    {
-      "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity",
-      "extension": [
-        {
-          "url": "ombCategory",
-          "valueCoding": {
-            "system": "urn:oid:2.16.840.1.113883.6.238",
-            "code": "2135-2",
-            "display": "Hispanic or Latino"
-          }
-        },
-        {
-          "url": "text",
-          "valueString": "Reported Unknown:Italian"
-        }
-      ]
-    },
-    {
-      "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
-      "extension": [
-        {
-          "url": "http://terminology.hl7.org/CodeSystem/v3-NullFlavor",
-          "valueCoding": {
-            "system": "http://terminology.hl7.org/CodeSystem/v3-NullFlavor",
-            "code": "UNK",
-            "display": "Unknown"
-          }
-        },
-        {
-          "url": "text",
-          "valueString": "Reported Unknown:Italian"
-        }
-      ]
-    },
-    {
-      "url": "http://fhir.ncpi-project-forge.io/StructureDefinition/age-at-event",
-      "valueAge": {
-        "value": 31.0,
-        "unit": "a",
-        "system": "http://unitsofmeasure.org",
-        "code": "years"
-      }
-    }
-  ],
-  "gender": "male"
-}
 
-def create_patient(host):
-    return host.post('Patient', example_patient, validate_only=False)
+@pytest.fixture(scope="module")
+def prep_server(host):
+    # For the CRUD, we just need to make sure we delete any leftover data from
+    # failed tests and then, in the teardown, we do it once again
+
+    test_data = load_test_data()
+    # We'll just walk through the _ALL for each resource type
+
+    for resource, record in test_data["_ALL"].items():
+        delete_content_for_resource(host, resource, record)
+
+    yield test_data
 
-def reset_crud_test_data(host):
-  pass
\ No newline at end of file
+    # And once again to clean up after this module completes
+    for resource, record in test_data["_ALL"].items():
+        delete_content_for_resource(host, resource, record)
```

Comment:
Are these lines lingering?

Reply:
I think the diff is throwing you off. Those lines belong after the yield as part of prep_server to delete the records after the test completes.

Reply:
Oh yeah, I mean, deletion was already executed and I wonder what this loop is for.